arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -239,9 +241,36 @@ uint64_t RocksDBMetaCollection::recalculateCounts() {
     snapshot = engine->db()->GetSnapshot();
     TRI_ASSERT(snapshot);
   }
```

Comment:
I'd like to suggest the following sequence:
1. Acquire the lock.
2. Generate a transaction ID.
3. Get the latest sequence number from the db.
4. Place a blocker using that transaction ID and sequence number.
5. Get a snapshot from the db.
6. Get the number of documents.
7. Release the lock.